scene/gui/line_edit.cpp

```diff
@@ -645,7 +645,9 @@ void LineEdit::_notification(int p_what) {
 			Color font_color_selected = get_color("font_color_selected");
 			Color cursor_color = get_color("cursor_color");
 
-			const String &t = text.empty() ? placeholder : text;
+			bool using_placeholder = text.empty();
+
+			const String &t = using_placeholder ? placeholder : text;
 			// draw placeholder color
 			if (text.empty())
```

Comment:
Maybe use `using_placeholder` here too for consistency.

Reply:
You can even define `using_placeholder` earlier in the callback and change `int cached_text_width = text.empty() ? cached_placeholder_width : cached_width;` accordingly (further up).